packages/mailchimp/src/Adaptor.js

```diff
@@ -312,6 +316,108 @@ export function listAudienceInfo(query, callback = s => s) {
   };
 }
 
+// Default options for request()
+const defaultOptions = {
+  query: {},
+  body: undefined,
+};
+
+// Assert response
+const assertOK = (response, fullUrl) => {
+  if (response.statusCode >= 400) {
+    const defaultErrorMesssage = `Request to ${fullUrl} failed with status: ${response.statusCode}`;
+
+    const error = new Error(defaultErrorMesssage);
+    error.code = response.statusCode;
+    error.url = fullUrl;
+    throw error;
+  }
+};
+/**
+ * Make an HTTP request to Mailchimp API
+ * @example <caption>Get list to all other resources available in the API</caption>
+ * request('GET','/');
+ * @example <caption>Create a new account export in your Mailchimp account</caption>
+ * request('POST','/accounts-export', {include_stages:[]});
+ * @function
+ * @param {string} method - The HTTP method for the request (e.g., 'GET', 'POST', 'PUT', 'DELETE').
+ * @param {string} path - The endpoint of the api to which the request should be made.
+ * @param {Object} options - Additional options for the request (query, body only).
+ * @param {function} [callback] - (Optional) callback function to handle the response.
+ * @returns {Operation}
+ */
+export const request = (method, path, options, callback = s => s) => {
+  return async state => {
+    const apiVersion = '3.0';
+    const { apiKey, server } = state.configuration;
+
+    const [resolvedMethod, resolvedPath, resolvedOptions] = expandReferences(
+      state,
+      method,
+      path,
+      options
+    );
+    const { query, body } = { ...defaultOptions, ...resolvedOptions };
+
+    const apiToken = Buffer.from(`openfn:${apiKey}`, 'utf-8').toString(
+      'base64'
+    );
+
+    const headers = {
+      'Content-Type': 'application/json',
+      Authorization: `Basic ${apiToken}`,
+    };
+
+    const urlPath = `/${apiVersion}${resolvedPath}`;
+    const response = await state.apiClient.request({
+      method: resolvedMethod,
+      path: urlPath,
+      headers,
+      query,
+      body: body ? JSON.stringify(body) : undefined,
+    });
+
+    assertOK(response, `https://${server}.api.mailchimp.com${urlPath}`);
+
+    const responseBody = await response.body.json();
+
+    const nextState = {
+      data: responseBody,
+      response: responseBody,
+    };
+    if (callback) return callback(nextState);
```

Comment:
There will always be a callback because you've defaulted it in the paramters

Reply:
✅ 👍🏽  [cf1ac78](https://github.com/OpenFn/adaptors/pull/352/commits/cf1ac7886f49a949686770b63a00f4487d175099)